app/Http/Controllers/ProfileController.php

```diff
@@ -4,40 +4,93 @@
+                return response()->json(['message' => 'User has no permission to view this profile'], 403);
+            }
+        }
+        return response()->json($addressProfile);
     }
 
-    public function updateMyProfile(ProfileRequest $request) {
+    public function updateMyProfile(ProfileRequest $request): JsonResponse
+    {
         return response()->json($this->repo->saveProfile($request));
     }
 
-    public function uploadProfileAvatar(ProfileUploadRequest $request, $address) : JsonResponse {
+    public function uploadMyProfileAvatar(ProfileUploadRequest $request): JsonResponse
+    {
+        $profile = $this->repo->uploadProfileAvatar($request);
+        if ($profile)
+            return response()->json($profile);
+
+        return response()->json(['message' => 'File Upload Failed', 422]);
+    }
+
+    public function uploadMyProfileBackground(ProfileUploadRequest $request): JsonResponse
+    {
+        $profile = $this->repo->uploadProfileBackground($request);
+        if ($profile)
+            return response()->json($profile);
+
+        return response()->json(['message' => 'File Upload Failed', 422]);
+    }
+
+    // to be deprecated
+    public function uploadProfileAvatar(ProfileUploadRequest $request, $address): JsonResponse
+    {
         $profile = $this->repo->uploadProfileAvatar($request, $request->address);
-        if($profile)
+        if ($profile)
             return response()->json($profile);
 
-        return response()->json(['message' => 'File Upload Failed' ,422]);
+        return response()->json(['message' => 'File Upload Failed', 422]);
     }
 
-    public function uploadProfileBackground(ProfileUploadRequest $request, $address) : JsonResponse {
-       $profile = $this->repo->uploadProfileBackground($request, $request->address);
-       if($profile)
-           return response()->json($profile);
+    // to be deprecated
+    public function uploadProfileBackground(ProfileUploadRequest $request, $address): JsonResponse
+    {
+        $profile = $this->repo->uploadProfileBackground($request, $request->address);
+        if ($profile)
+            return response()->json($profile);
 
-        return response()->json(['message' => 'File Upload Failed' ,422]);
+        return response()->json(['message' => 'File Upload Failed', 422]);
     }
 
+    public function manifest(Request $request): JsonResponse
+    {
+        $profile = $this->repo->getProfile($request->get('address'), ['users.circle.epoches', 'users.circle.nominees']);
+        if (!$profile && count($profile->users))
+            abort('403', 'You do not have an active account in Coordinape');
+
+        if ($profile->admin_view) {
+            $circle_ids = ['*'];
+        } else {
+            $circle_ids = $profile->users->pluck('circle_id')->toArray();
+        }
+        $profile->tokens()->delete();
+        $token = $profile->createToken('circle-access-token', $circle_ids)->plainTextToken;
```

Comment:
guess its because of the if statement is not checking whether the profile exists correctly
 ```
  if (!$profile && count($profile->users))
    abort('403', 'You do not have an active account in Coordinape');
```
replaced to this
https://github.com/coordinape/coordinape-backend/blob/5f3828cd02e0cc3265d60f64b1dc3a1f1987ee61/app/Http/Controllers/ProfileController.php#L78